scripts/generate_pymol_segmentation.py

```diff
@@ -0,0 +1,85 @@
+######
+# Generates a pymol script to visualize the segmentation from the standardized chopping information
+# The residues which are not part of any domain are colored in gray80
+# The script can be run from PyMol using the command: run <protein.pml> or by going to File > Run Script
+
+# Input: <protein.tsv> (tsv file from collect_segmentation_intermediates.py)
+#        <protein.pdb>
+#        <method> (chainsaw, k_geostas, h_geostas)
+#        <k> (1-10)
+#        <protein.pml> (optional)
+# Output: <protein.pml>
+######
+
+
+import sys
+import os
+from pathlib import Path
+import csv
+
+if len(sys.argv) < 3:
+    print(f"USAGE: python {os.path.basename(__file__)} <protein.tsv> <protein.pdb> <method> <k> [protein.pml]")
+    sys.exit(1)
+
+tsv_file = sys.argv[1]
+pdb_file = sys.argv[2]
+method = sys.argv[3]
+k = sys.argv[4]
+output_file = sys.argv[5] if len(sys.argv) > 5 else (f'{Path(pdb_file).stem}.pml')
+
+
+def parse_chopping(chopping, method, k):
+    """
+    Columns: 'index', 'method', 'domain_count', 'chopping'
+    Example chopping format: '1-100,101-200,201-300_301-400,401-500'
+    Method choices: 'chainsaw', 'k_geostas', 'h_geostas'
+    k choices: 2-10
+    """
+    methods = {'chainsaw': 'chainsaw',
+               'k_geostas': 'GeoStaS K-means',
+               'h_geostas': 'GeoStaS Hierarchical'}
+    segments = {}
+    with open(tsv_file, 'r') as f:
+        # Skip the header
+        reader = csv.DictReader(f, delimiter='\t')
+        for row in reader:
+            if methods[method] in row['method'] and row['domain_count'] == k:
+                chopping = row['chopping']
+                break
+        if chopping == 'NA':
+                print(f"Chopping not available for {method} with k={k}")
+                exit(1)
+        for i, domain in enumerate(chopping.split(',')):
+            r = []
+            subdomains = domain.split('_')
+            for subdomain in subdomains:
+                start, end = subdomain.split('-')
```

Comment:
If you call the script with `... chainsaw 12 ...` or something, a valid value for method, but an invalid one for K, this line blows up, because it reaches there in the first place. A solution might be to check if `chopping is None` and `exit(1)` in that case.

I'd also consider instead of exiting, to return from the function. It's a bit surprising when a function that is expected to return a value exits the script instead. So, for instance, you could check the result of `parse_chopping` and if it's empty, print an error and exit 1 at that point. But this can be fixed later as we pull these scripts out into classes.

Reply:
No real reason, it does make it more simple to manipulate once you have the protein in Pymol so I changed it to selections 👍

Reply:
Move the exit to the end of the script instead but like you said it can be changed in the class in the future.